packages/extensions/openapi-to-typespec/src/utils/library-type-mapping.ts

```diff
@@ -0,0 +1,250 @@
+import { ChoiceSchema, isObjectSchema, Schema, SealedChoiceSchema } from "@autorest/codemodel";
+import { getArmCommonTypeVersion } from "../autorest-session";
+import { isArraySchema, isChoiceSchema, isDictionarySchema, isSealedChoiceSchema, isUuidSchema } from "./schemas";
+
+// TO-DO: we might need a isReadonly
+interface LibraryPropertyBase {
+  serializedName: string;
+  required: boolean;
+  type: string;
+}
+
+interface LibraryPrimitiveProperty extends LibraryPropertyBase {
+  type: "primitive";
+  schema: (schema: Schema) => boolean;
+}
+
+interface LibraryNonPrimitiveProperty extends LibraryPropertyBase {
+  type: "nonPrimitive";
+  schema: () => LibraryType;
+}
+
+type LibraryProperty = LibraryPrimitiveProperty | LibraryNonPrimitiveProperty;
+
+interface LibraryTypeBase {
+  name: string;
+  type: string;
+}
+
+interface LibraryObjectType extends LibraryTypeBase {
+  type: "object";
+  properties: Record<string, LibraryProperty>;
+}
+
+interface LibraryArrayType extends LibraryTypeBase {
+  type: "array";
+  elementType: () => LibraryType;
+}
+
+interface LibraryPrimitiveType extends LibraryTypeBase {
+  type: "primitive";
+  schema: (schema: Schema) => boolean;
+}
+
+interface LibraryEnumType extends LibraryTypeBase {
+  type: "enum";
+  sealed: boolean;
+  values: Record<string, string>;
+}
+
+interface LibraryDictionaryType extends LibraryTypeBase {
+  type: "dictionary";
+  valueType: () => LibraryType;
+}
+
+export type LibraryType =
+  | LibraryObjectType
+  | LibraryEnumType
+  | LibraryArrayType
+  | LibraryPrimitiveType
+  | LibraryDictionaryType;
+
+export function isEquivalent(schema: Schema, expected: LibraryType): boolean {
+  if (expected.type === "object") {
+    if (!isObjectSchema(schema)) return false;
+    if (schema.properties?.length !== Object.keys(expected.properties).length) return false;
+
+    for (const property of schema.properties ?? []) {
+      if (property.serializedName in expected.properties) {
+        const expectedProperty = expected.properties[property.serializedName];
+        if ((property.required ?? false) !== expectedProperty.required) return false;
+
+        if (expectedProperty.type === "primitive") {
+          if (!expectedProperty.schema(property.schema)) return false;
```

Comment:
seems like the schema function is always property.schema.type == expectedProperty.name?
can we get rid of the schema function?